xarray/tests/test_coding.py

```diff
@@ -105,3 +105,15 @@ def test_scaling_converts_to_float32(dtype):
     roundtripped = coder.decode(encoded)
     assert_identical(original, roundtripped)
     assert roundtripped.dtype == np.float32
+
+
+@pytest.mark.parametrize("scale_factor", (10, [10]))
+@pytest.mark.parametrize("add_offset", (0.1, [0.1]))
+def test_scaling_offset_as_list(scale_factor, add_offset):
+    # test for #4631
+    encoding = dict(scale_factor=scale_factor, add_offset=add_offset)
+    original = xr.Variable(("x",), np.arange(10.0), encoding=encoding)
+    coder = variables.CFScaleOffsetCoder()
+    encoded = coder.encode(original)
+    roundtripped = coder.decode(encoded)
+    assert_allclose(original, roundtripped)
```

Comment:
Unless I'm missing something, this test just checks that a round trip is successful. It doesn't check that encoding with `scale_factor` or `add_offset` specified as a list (a) is equivalent to encoding with a scalar, or (b) is correct. An implementation that simply ignored lists for `scale_factor` or `add_offset` (other than retaining the `encoding` dictionary) would pass this test.

Reply:
Good point!